screenshot-testing/src/main/java/com/stripe/android/screenshottesting/RenderSessionCleaner.kt

```diff
@@ -0,0 +1,130 @@
+    if (weakToRunTrampolined != null) {
+        val toRunTrampolined = weakToRunTrampolined.get()
+        toRunTrampolined?.clear()
+    }
+    dispose()
+}
+
+/**
+ * Performs dispose() call against View object associated with [ViewInfo] if that object is an
+ * instance of Paparazzi's internal `ComposeViewAdapter` instance.
+ *
+ * @param viewInfo a [ViewInfo] associated with the View object to be potentially disposed of
+ */
+private fun disposeIfCompose(viewInfo: ViewInfo) {
+    val viewObject: Any? = viewInfo.viewObject
+    if (viewObject?.javaClass?.name != PAPARAZZI_COMPOSE_VIEW_ADAPTER_FQN) {
+        return
+    }
+    try {
+        val composeView = viewInfo.children[0].viewObject as ComposeView
+        composeView.disposeComposition()
+    } catch (ex: IllegalAccessException) {
+        LOGGER.warning("Unexpected error while disposing compose view $ex")
+    } catch (ex: InvocationTargetException) {
+        LOGGER.warning("Unexpected error while disposing compose view $ex")
+    }
+}
+
+private fun findApplyObservers(): MutableCollection<*>? {
+    try {
+        val applyObserversField = Class.forName(SNAPSHOT_KT_FQN).getDeclaredField("applyObservers")
+        applyObserversField.isAccessible = true
+        val applyObservers = applyObserversField[null]
+        if (applyObservers is MutableCollection<*>) {
+            return applyObservers
+        }
+        LOGGER.warning("SnapshotsKt.applyObservers found but it is not a List")
+    } catch (ex: ReflectiveOperationException) {
+        LOGGER.warning("Unable to find SnapshotsKt.applyObservers $ex")
+    }
+    return null
+}
+
+private fun findToRunTrampolined(): MutableCollection<*>? {
+    try {
+        val uiDispatcher = AndroidUiDispatcher::class.java
+        val uiDispatcherCompanion = AndroidUiDispatcher.Companion::class.java
+        val uiDispatcherCompanionField = uiDispatcher.getDeclaredField("Companion")
+        val uiDispatcherCompanionObj = uiDispatcherCompanionField[null]
+        val getMainMethod =
+            uiDispatcherCompanion.getDeclaredMethod("getMain").apply { isAccessible = true }
+        val mainObj = getMainMethod.invoke(uiDispatcherCompanionObj)
+        val combinedContext = Class.forName(COMBINED_CONTEXT_FQN)
+        val elementField = combinedContext.getDeclaredField("element").apply { isAccessible = true }
+        val uiDispatcherObj = elementField[mainObj]
+
+        val toRunTrampolinedField =
+            uiDispatcher.getDeclaredField("toRunTrampolined").apply { isAccessible = true }
+        val toRunTrampolinedObj = toRunTrampolinedField[uiDispatcherObj]
+        if (toRunTrampolinedObj is MutableCollection<*>) {
+            return toRunTrampolinedObj
+        }
+        LOGGER.warning("AndroidUiDispatcher.toRunTrampolined found but it is not a MutableCollection")
+    } catch (ex: ReflectiveOperationException) {
+        LOGGER.warning("Unable to find AndroidUiDispatcher.toRunTrampolined $ex")
+    }
+    return null
+}
```

Comment:
This was pulled & modified from the [Paparazzi GitHub issue](https://github.com/cashapp/paparazzi/issues/915#issuecomment-1783114569) in regards to this memory leak with `layoutlib` and `Jetpack Compose` that is heavily based on [Google's own solution](https://issuetracker.google.com/issues/290990640#comment5) for the problem in regards to Android Studio.

Reply:
I think this is worth leaving as a comment in code. That way if something changes we can easily reference back.

Reply:
Left as a top-level comment for the `dispose` function!